src/bullet.rs

```diff
@@ -11,40 +9,45 @@ impl Plugin for BulletPlugin {
     }
 }
 
-pub fn bullet_damage(mut enemies: Query<&mut Health>, mut events: EventReader<CollisionEvent>) {
-    let bullet_collisions = events
+pub fn bullet_damage(
+    mut entities: Query<(&mut Health, &ChickenOrDog)>,
+    bullets: Query<(&Collisions, &Bullet)>,
+) {
+    // have to collect the iter to drop the entities query
+    #[allow(clippy::needless_collect)]
+    let entities_to_damage = bullets
         .iter()
-        .filter(|e| e.is_started())
-        .filter_map(is_bullet_collision);
-    for (_bullet, entity) in bullet_collisions {
-        if let Ok(mut health) = enemies.get_mut(entity) {
-            health.0 -= 1.0;
-        }
+        .flat_map(|(collisions, bullet)| {
+            collisions
+                .entities()
+                .map(move |collision| (collision, bullet))
+        })
+        .filter_map(|(entity, bullet)| {
+            let (_, entity_team) = entities.get(entity).ok()?;
+            if bullet.origin_team != *entity_team {
+                Some(entity)
+            } else {
+                None
+            }
+        })
```

Comment:
hmm thats strange, can we not clone the entity or something

Reply:
no and actually `Entity` is returned from the iter. We have to close out the reading of entities before we can mutate any.